snapd.go

```diff
@@ -221,6 +221,10 @@ func action(ctx *cli.Context) {
 	// same variables in that configuration
 	applyCmdLineFlags(cfg, ctx)
 
+	// Switch log level to user defined
+	log.Info("setting log level to: ", l[cfg.LogLevel])
+	log.SetLevel(getLevel(cfg.LogLevel))
```

Comment:
This validates tat buffering until a Fatal, exit, or the settings are done is a better path. It would honor both the log level and the log output.